### DISCUSSION TITLE
REST API CRUD

### DISCUSSION THREAD

**Discussion by prabha-ar** [microfeed/microfeed#124] [General]
Hi everyone I am building a citizen run NEWS app .. using MICROFEED as CMS thorough REST API

How does this REST CRUD operations work, does it uses cloudflare workers?
will it handle a large write operations at a time, is there any limit ?

**Comment by wenbinf** [comment]
`How does this REST CRUD operations work, does it uses cloudflare workers?`

The whole backend code runs on Cloudflare Pages.
API belongs to backend code.
Cloudflare Pages runs on top of Cloudflare Workers.

So yes, API runs on top of Cloudflare Workers.

You can find the API code here: https://github.com/microfeed/microfeed/tree/main/functions/api

`will it handle a large write operations at a time, is there any limit ?`

For typical write operations, it should be okay. For example, you can use it to publish a long blog post.
If you'd like to upload an image / audio / video file, then you are actually using a presigned R2 url and there's a limit for file size (2GB?)
